src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,544 @@
+    Gnew = groebner_lift(Hnew,adjacentOrdering,G,ordG)
+    return Gnew,adjacentOrdering
+
+end
+# helper functions for groebner_flip
+function groebner_flip_adjacent_ordering(R::MPolyRing,
+                                         homogeneityVector::Vector{fmpz},
+                                         interior_facet_point::Vector{fmpz},
+                                         outer_normal_vector::Vector{fmpz})
+
+    return weight_ordering(Int.(homogeneityVector),
+                           weight_ordering(Int.(interior_facet_point),
+                                           weight_ordering(Int.(outer_normal_vector),
+                                                           revlex(R))))
+
+end
+function groebner_flip_adjacent_ordering(R::MPolyRing,
+                                         homogeneityVector::Nothing,
+                                         interior_facet_point::Vector{fmpz},
+                                         outer_normal_vector::Vector{fmpz})
+
+    return weight_ordering(Int.(interior_facet_point),
+                           weight_ordering(Int.(outer_normal_vector),
+                                           revlex(R)))
+
+end
+
+
+
+# constructs a polyhedral fan from list of cones
+function polyhedral_fan_from_cones(listOfCones::Vector{<:Cone})
+
+    ###
+    # Collect incidence information
+    ###
+    fanRays = Vector{Vector{fmpq}}()
+    fanIncidences = Vector{Vector{Int}}()
+    for cone in listOfCones
+        coneIncidence = Vector{Int}()
+        coneRays,_ = rays_modulo_lineality(cone)
+        for coneRay in coneRays
+            # if ray does not occur in rayList, add it to it
+            # either way add the corresponding index to rayIncidences
+            i = findfirst(isequal(coneRay),fanRays)
+            if i === nothing
+                push!(fanRays,coneRay)
+                push!(coneIncidence,length(fanRays))
+            else
+                push!(coneIncidence,i)
+            end
+        end
+        push!(fanIncidences,coneIncidence)
+    end
+
+    # convert Vector{Vector} to IncidenceMatrix and Matrix respectively
+    fanIncidences = IncidenceMatrix(fanIncidences)
+    fanRays = matrix(QQ,fanRays)
+
+    fanLineality = matrix(QQ,lineality_space(first(listOfCones)))
+
+    return PolyhedralFan(fanRays,fanLineality,fanIncidences)
+end
+
+
+
+
+@doc raw"""
+    groebner_fan(I::MPolyIdeal; return_groebner_bases::Bool=false, return_orderings::Bool=false, verbose_level::Int=0)
+
+# Input
+- `I`: A multivariate polynomial ideal over a coefficient field
+- `return_groebner_bases`: If `true`, return all Groebner bases computed during the traversal
+- `return_orderings`: If `true`, return all monomial orderings used during the traversal
+- `verbose_level`: If positive, print out how many cones have been computed during the traversal
+
+# Output
+A `PolyhedralFan` that is the Groebner fan of `I`.
```

Comment:
This Input/Output format is discouraged by Julia, and I agree with that.  It should ideally only be used if it just can't be reasonably avoided. But here I think it can easily be avoided? E.g. like this:
```suggestion
Return a `PolyhedralFan` representing the Groebner fan of `I`, where is a multivariate polynomial ideal over a field.

If `verbose_level` is positive, also print how many cones have been computed during the traversal.
```
The description of the `return_*` kwargs seems to be redundant with what follows?

BTW also `verbose_level` should perhaps be removed and instead replaced with use of `@vprint` and friends -- but that could be done in a future PR, of course